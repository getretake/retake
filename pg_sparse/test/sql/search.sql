SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;
CREATE INDEX idx_mock_items ON mock_items USING sparse_hnsw(sparse_embedding);
SET enable_seqscan = off;
SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;
DROP INDEX idx_mock_items;
CREATE INDEX idx_mock_items ON mock_items USING sparse_hnsw(sparse_embedding) WITH (ef_search=64, ef_construction=64, m=16);
SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;

