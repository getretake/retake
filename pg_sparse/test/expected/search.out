SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;
        description        
---------------------------
 Comfortable slippers
 Organic green tea
 Interactive board game
 Paperback romantic novel
 Bluetooth-enabled speaker
(5 rows)

CREATE INDEX idx_mock_items ON mock_items USING sparse_hnsw(sparse_embedding);
SET enable_seqscan = off;
SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;
        description        
---------------------------
 Organic green tea
 Paperback romantic novel
 Interactive board game
 Comfortable slippers
 Bluetooth-enabled speaker
(5 rows)

DROP INDEX idx_mock_items;
CREATE INDEX idx_mock_items ON mock_items USING sparse_hnsw(sparse_embedding) WITH (ef_search=64, ef_construction=64, m=16);
SELECT description FROM mock_items ORDER BY sparse_embedding <==> '[0,0,0,3,0,2,0,4,0,1]' LIMIT 5;
        description        
---------------------------
 Organic green tea
 Paperback romantic novel
 Interactive board game
 Comfortable slippers
 Bluetooth-enabled speaker
(5 rows)

